### ISSUE TITLE
File associations and URL handlers

### ISSUE THREAD

**Issue by mikehearn** [hydraulic-software/conveyor#5]
Vote for this if you need packaged apps to support registering file associations and URL handlers.

**Comment by mikehearn** [comment]
URL handlers can be done in Conveyor 4 using some custom config. Future releases will simplify the configuration required, but that isn't strictly necessary. See here for how:

https://conveyor.hydraulic.dev/4.0/configs/os-integration/#url-handlers

**Comment by ennerf** [comment]
The MSIX also works for [file associations](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-extensions#example-4)

```xml
  manifests.msix.extensions-xml = """
  <uap:Extension Category="windows.fileTypeAssociation">
      <uap3:FileTypeAssociation Name="textfile" Parameters="%1">
        <uap:SupportedFileTypes>
          <uap:FileType>.txt</uap:FileType>
        </uap:SupportedFileTypes>
      </uap3:FileTypeAssociation>
  </uap:Extension>
  """
```

**Comment by ennerf** [comment]
The actual file handling on macOS is not straight forward though. Windows and Linux supply the file name as the first argument, but macOS needs to integrate with a native library.

@mikehearn I saw you started [How to handle file open requests on MacOS](https://openjfx-dev.openjdk.java.narkive.com/emgVmZMH/how-to-handle-file-open-requests-on-macos) 8 years ago. Do you by any chance happen to know of a solution that is compatible with JavaFX 20?

**Comment by mikehearn** [comment]
Yes, that looks right. We're adding native support for file associations at the moment. It won't be in the next feature release, probably the one after that. It boils down to what you're saying plus some code to handle Linux.

**Comment by mikehearn** [comment]
Re: JavaFX. Yes, it needs some more work on the native side to do the equivalent for AWT.

**Comment by ennerf** [comment]
Nice 👍  Is there a way to bundle icons too?

**Comment by ennerf** [comment]
I didn't see icons mentioned in the docs, so I wasn't sure what it does without trying. Registering the application icon makes sense and would work well enough for me 👍

**Comment by mikehearn** [comment]
Yep, the icon used is the application icon.

**Comment by ennerf** [comment]
Fyi, I just replaced my previous hacks and it works like a charm 👍 

The only issue I ran into was that Ubuntu 20.04 does not show the icon for the mime-type, in case it should be doing that.
